File: Tests/FocusCoreTests/FocusSettingsTests.swift
import FocusCore
import Foundation
import Testing

struct FocusSettingsTests {
    @Test
    func defaultsMatchExpectedContract() {
        let settings = FocusSettings.defaults

        #expect(settings.redirectEnabled)
        #expect(settings.playerMaskEnabled)
        #expect(settings.searchMaskEnabled)
        #expect(settings.dynamicMaskEnabled)
        #expect(!settings.debugMode)
        #expect(settings.defaultEntry == .dynamic)
    }

    @Test
    func settingsPersistAndReload() {
        let suiteName = "FocusSettingsTests.settingsPersistAndReload"
        let defaults = UserDefaults(suiteName: suiteName)!
        defaults.removePersistentDomain(forName: suiteName)

        let settings = FocusSettings(
            redirectEnabled: false,
            playerMaskEnabled: false,
            searchMaskEnabled: true,
            dynamicMaskEnabled: false,
            debugMode: true,
            defaultEntry: .search
        )

        settings.save(to: defaults)

        #expect(FocusSettings.load(from: defaults) == settings)
    }
}

File: focus-core/src/commonTest/kotlin/org/bilibilifocus/core/model/FocusSettingsTests.kt
package org.bilibilifocus.core.model

import org.bilibilifocus.core.storage.SettingsRepository
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class FocusSettingsTests {

    @Test
    fun `defaults match expected contract`() {
        val settings = FocusSettings.DEFAULTS

        assertTrue(settings.redirectEnabled)
        assertTrue(settings.playerMaskEnabled)
        assertTrue(settings.searchMaskEnabled)
        assertTrue(settings.dynamicMaskEnabled)
        assertFalse(settings.debugMode)
        assertEquals(FocusEntry.DYNAMIC, settings.defaultEntry)
    }

    @Test
    fun `serialization round-trip preserves values`() {
        val settings = FocusSettings(
            redirectEnabled = false,
            playerMaskEnabled = false,
            searchMaskEnabled = true,
            dynamicMaskEnabled = false,
            debugMode = true,
            defaultEntry = FocusEntry.SEARCH,
        )

        val json = kotlinx.serialization.json.Json.encodeToString(FocusSettings.serializer(), settings)
        val restored = kotlinx.serialization.json.Json.decodeFromString(FocusSettings.serializer(), json)

        assertEquals(settings, restored)
    }
}
